lib/config/flat-config-schema.js

```diff
@@ -323,7 +311,9 @@ const processorSchema = {
     merge: "replace",
     validate(value) {
         if (typeof value === "string") {
-            assertIsPluginMemberName(value);
+            if (!value.includes("/")) {
```

Comment:
This would allow specifiers like `"/"`, `"/foo"`, or `"foo/"`, which would then be parsed and processed as empty string plugin/processor names. I think we should check if there's at least one character before and after `/`.

Reply:
I think the original regex aimed to check that there's at least one non-slash character before, after, and between slash characters, which seems fine, it was just too restrictive on allowed characters.

Reply:
I tend to agree, but just to play out the corner cases, the config schema does not currently prevent me naming a plugin `""`, nor a processor `""`, so technically all of those are _currently_ valid object IDs.

If you wanted to tighten up the spec on what plugins/objects can be named, I certainly wouldn't disagree 😉

Reply:
> If you wanted to tighten up the spec on what plugins/objects can be named, I certainly wouldn't disagree

I think we should tighten up the spec because empty names can cause referencing ambiguities, and I think we should keep requiring non-empty names in processor references here.

Reply:
I was going to separately suggest some validation that prevents users from specifying plugins in a way that creates ambiguous object  ids, i.e. 2 different rules map to the same object ID.

Perhaps we could also limit plugin names to valid npm package names (for compat), and rule/processor names to alphanumeric and a handful of special characters?

I know this is going well beyond this issue now and will probably need an RFC etc., how would you like to move forward?

Reply:
I'm certainly not suggesting we widen this PR out beyond relaxing validation of `processor`. 😄

But it's not just about allowing dot-processors, it's about the poor UX if a user can't seem to reference a processor by ID, even if they've got the ID correct.

Renaming a processor in an external plugin would be a relatively advanced configuration thing to do, and it's not at all clear to a user that that's the solution to this problem.

I'd suggest loosening this as in the PR _for now_ for maximum ecosystem compatibility and better UX, and then follow up with looking at constraining plugin and object names, at which time we can tighten this regex back up?

Reply:
> I'd suggest loosening this as in the PR for now for maximum ecosystem compatibility and better UX, and then follow up with looking at constraining plugin and object names, at which time we can tighten this regex back up?

We can't loosen the restrictions and then tighten them up again later -- that's a breaking change. Anytime we change was it allowed we need to think deeply about it because it's very hard to make stricter later if we make a mistake.

There are already ways to work around the limitation. First, you can directly reference the processor:

```js
import plugin from "some-plugin";

export default [{
    plugins: { plugin }
    processor: plugin.processors[".txt"]
}];
```

The processor itself can also export the same processor with two names:

```js
const processor = {};

export default {
    processors: {
        ".txt": processor,
        other: processor
    }
};
```

So, I don't feel like we are putting people in any kind of difficult situation with how we currently restrict processor names.

And actually, through this conversation, I think I've arrived at the opinion that we should keep things as-is. To the extent that we want to discourage file-extension processors, the naming restriction for use in flat config seems like a good idea. I don't see a strong reason to make a change at this point.

Reply:
In fairness, re-tightening up the regex here _after_ restricting what plugins/objects can be named wouldn't be a breaking change, it would just be changing the error message (i.e. "cannot find `plugin/object` to `that's not a valid object ID`).

The breaking change would be restricting plugin/object names.

Restricting those names is not necessarily something I'd advocate for, but I think that if we are to leave them unrestricted, then it should be possible to consistently refer to any object by the ID formed from `${pluginName}/${objectName}`.

It's a bad user experience otherwise because they'll spend their time thinking they got the ID wrong somehow (because that's what ESLint is telling them) and trying to get it right when it's not clear that it's actually impossible to reference it by ID.

In principle, if I generate a list of all the processors in all the plugins like so:

```js
function *getAllProcessors(config) {
    for(const [pluginName, plugin] in Object.entries(config.plugins ?? {})) {
        for (const [processorName, processor] in Object.entries(plugin.processors ?? {})) {
            yield $`${pluginName}/${processorName}`;
        }
    }
}
```

then I don't think it's unreasonable to say that I should be able to use any of those IDs in my config. If some of them are "invalid" IDs, then I shouldn't be allowed to name my plugin/processor in a way that generates an invalid ID. (this does ignore potentially ambiguous object IDs that it's currently possible to form, but that's a separate issue)